modules/setting/setting.go

```diff
@@ -1221,6 +1199,76 @@ func NewContext() {
 	}
 }
 
+func loadInternalToken(sec *ini.Section) string {
+	uri := sec.Key("INTERNAL_TOKEN_URI").String()
+	if len(uri) == 0 {
+		return loadOrGenerateInternalToken(sec)
+	}
+	tempURI, err := url.Parse(uri)
+	if err != nil {
+		log.Fatal(4, "Failed to parse INTERNAL_TOKEN_URI (%s). Falling back to INTERNAL_TOKEN: %v", uri, err)
```

Comment:
Perhaps I am misunderstanding, but wouldn't it be incorrect to say "Falling back to INTERNAL_TOKEN" since Fatal actually hard-stops? It looks as thought it actually falls back to INTERNAL_TOKEN a few lines above if the URI doesn't exist.

Reply:
Yes that should be warning at most

Reply:
If it is set to `Warning` you will get a dupe log since it will then fall through the switch-case. (Or it may even panic, since `tempURI` failed to parse)  
Just removing the `Falling back to INTERNAL_TOKEN` bit would suffice, perhaps adding a log statement above as well where it does fall back to INTERNAL_TOKEN.

Reply:
Thanks for feedback :) I've removed mention about fallback because you are correct in that it doesn't fallback.